File: 2452-words-within-two-edits-of-dictionary/2452-words-within-two-edits-of-dictionary.ts
function twoEditWords(queries: string[], dictionary: string[]): string[] {
    const ans: string[] = [];

    for (const q of queries) {
        for (const d of dictionary) {
            let diff = 0;

            for (let i = 0; i < q.length; i++) {
                if (q[i] !== d[i]) {
                    diff++;
                }

                if (diff > 2) break;
            }

            if (diff <= 2) {
                ans.push(q);
                break;
            }
        }
    }

    return ans;
}

File: 2452-words-within-two-edits-of-dictionary/2452-words-within-two-edits-of-dictionary.cs
public class Solution {
    public IList<string> TwoEditWords(string[] queries, string[] dictionary) {
        List<string> ans = new List<string>();

        foreach (string q in queries) {
            foreach (string d in dictionary) {
                int diff = 0;

                for (int i = 0; i < q.Length; i++) {
                    if (q[i] != d[i]) {
                        diff++;
                    }

                    if (diff > 2) break;
                }

                if (diff <= 2) {
                    ans.Add(q);
                    break;
                }
            }
        }

        return ans;
    }
}